nomad/structs/structs.go

```diff
@@ -134,6 +134,11 @@ const (
 )
 
 const (
+	// SystemInitializationType is used for messages that initialize parts of
+	// the system, such as the state store. These messages are not included in
+	// the event stream.
+	SystemInitializationType MessageType = 127
```

Comment:
I didn't know how to call this, so I'm open to suggestions 😅 

I wanted to keep it somewhat generic because `MessageType` is an `uint8`, so we don't have that many slots open. I also thought it would be interesting to have a `IgnoreEventStreamTypeFlag` that you could `&` to the appropriate message type, but we don't have any bits left since `IgnoreEventStreamTypeFlag` already uses 128.